src/main/java/com/mwt/contexts/SimpleContext.java

```diff
@@ -0,0 +1,29 @@
+package com.mwt.contexts;
+
+import com.mwt.misc.Feature;
+
+import java.util.ArrayList;
+
+/**
+ * A sample context class that stores a vector of Features.
+ */
+public class SimpleContext {
+  private ArrayList<Feature> features;
+
+  public SimpleContext(ArrayList<Feature> features) {
+    this.features = features;
+  }
+
+  public ArrayList<Feature> getFeatures() {
+    return features;
+  }
+
+  @Override
+  public String toString() {
+    String out = "";
+    for (Feature feature: features) {
+      out = out + feature.getId() + ": " + feature.getValue() + " ";
```

Comment:
Mmm I'm not particularly concerned about the performance of this unless features is large and you're calling this a huge number of times, but it would be better practice to use stringbuilder here because I'm appending in a loop, so I'll make that change.